changelog.d/13575.misc

```diff
@@ -0,0 +1 @@
+Optimize how Synapse calculates domains to fetch from during backfill.
```

Comment:
Waiting to merge this until we get a little more time for our baseline metrics on `matrix.org`